admin/tests/test_integration.py

```diff
@@ -430,3 +433,100 @@ def test_sdconfig_enable_https_on_source_interface():
+    # Now we will put in our own git configuration
+    with io.open('../.git/config', 'w') as fobj:
+        fobj.write(GIT_CONFIG)
+    # Let us move to an older tag
+    subprocess.check_call('git checkout 0.6'.split())
+    yield tmpdir
+
+
+# This class is to test all the git related operations.
+class TestGitOperations:
+    def test_check_for_update_when_updates_needed(self, securedrop_git_repo):
+        cmd = os.path.join(os.path.dirname(CURRENT_DIR),
+                           'securedrop_admin/__init__.py')
+        ansible_base = os.path.join(str(securedrop_git_repo),
+                                    'securedrop/install_files/ansible-base')
+        fullcmd = 'python {0} --root {1} check_for_updates'.format(
+                  cmd, ansible_base)
+        child = pexpect.spawn(fullcmd)
+        child.expect('Update needed', timeout=20)
+
+        child.expect(pexpect.EOF, timeout=10)  # Wait for CLI to exit
+        child.close()
+        assert child.exitstatus == 0
+        assert child.signalstatus is None
+
+    def test_check_for_update_when_updates_not_needed(self,
+                                                      securedrop_git_repo):
+        # Determine latest production tag using GitHub release object
+        github_url = 'https://api.github.com/repos/freedomofpress/securedrop/releases/latest'  # noqa: E501
+        latest_release = requests.get(github_url).json()
+        latest_tag = str(latest_release["tag_name"])
+
+        subprocess.check_call(["git", "checkout", latest_tag])
+
+        cmd = os.path.join(os.path.dirname(CURRENT_DIR),
+                           'securedrop_admin/__init__.py')
+        ansible_base = os.path.join(str(securedrop_git_repo),
+                                    'securedrop/install_files/ansible-base')
+        fullcmd = 'python {0} --root {1} check_for_updates'.format(
+                  cmd, ansible_base)
+        child = pexpect.spawn(fullcmd)
+        child.expect('All updates applied', timeout=20)
+
+        child.expect(pexpect.EOF, timeout=10)  # Wait for CLI to exit
+        child.close()
+        assert child.exitstatus == 0
+        assert child.signalstatus is None
+
+    def test_update(self, securedrop_git_repo):
+        gpgdir = os.path.join(os.path.expanduser('~'), '.gnupg')
+        os.mkdir(gpgdir)
+
+        # Set a reliable default keyserver for the tests.
+        with open(os.path.join(gpgdir, 'gpg.conf'), 'a') as f:
+            f.write('keyserver hkp://pgp.mit.edu')
+
+        # Ensure correct permissions on .gnupg home directory.
+        os.chmod(gpgdir, 0700)
+
+        cmd = os.path.join(os.path.dirname(CURRENT_DIR),
+                           'securedrop_admin/__init__.py')
+        ansible_base = os.path.join(str(securedrop_git_repo),
+                                    'securedrop/install_files/ansible-base')
+        child = pexpect.spawn('python {0} --root {1} update'.format(
+                              cmd, ansible_base))
```

Comment:
Non blocking remark: `make -C admin test` display

<pre>
securedrop_admin/__init__.py     395     75    81%   535, 540-547, 557-562, 569-581, 586-596, 601-603, 706-714, 718-727, 731-783, 787-801, 805
</pre>
which reports 787-801 i.e. the `main` function is not covered. The python command should be `coverage run python` to accumulate coverage information instead.